### PR TITLE
feat(scopes): apply scopes on leaderboard charts

### PR DESCRIPTION
**Pull request by larcin**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Create automatic tests
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: src/services/leaderboard-stat-getter.js [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
  */
-function LeaderboardStatGetter(model, modelRelationship, params) {
+function LeaderboardStatGetter(childModel, parentModel, params, user) {
   const labelField = params.label_field;
   const aggregate = params.aggregate.toUpperCase();
   const { limit } = params;
-  const schema = Schemas.schemas[model.name];
-  const schemaRelationship = Schemas.schemas[modelRelationship.name];
+  const schema = Schemas.schemas[childModel.name];
+  const schemaRelationship = Schemas.schemas[parentModel.name];
```
Comment (romain-gilliotte) [inline src/services/leaderboard-stat-getter.js:38]: why not keep calling those parent/child?

File: src/services/leaderboard-stat-getter.js [hunk 2]
```diff
@@ -52,26 +60,32 @@ function LeaderboardStatGetter(model, modelRelationship, params) {
...
+      where: parentSequelizeOptions.where,
+      includeIgnoreAttributes: false,
+      include: [{
+        model: childModel,
+        attributes: [labelField],
+        as: associationAs,
+        required: true,
+        where: childSequelizeOptions.where,
+        include: childSequelizeOptions.include || [],
+      }, ...(parentSequelizeOptions.include || [])],
```
Comment (romain-gilliotte) [inline src/services/leaderboard-stat-getter.js:82]: Did you test when there are duplication (the same model twice)?
What happens on the generated query?

File: src/services/leaderboard-stat-getter.js [hunk 3]
```diff
@@ -52,26 +60,32 @@ function LeaderboardStatGetter(model, modelRelationship, params) {
...
+        [childModel.sequelize.fn(aggregate, childModel.sequelize.col(aggregateField)), 'value'],
+      ],
+      where: parentSequelizeOptions.where,
+      includeIgnoreAttributes: false,
+      include: [{
+        model: childModel,
+        attributes: [labelField],
+        as: associationAs,
+        required: true,
+        where: childSequelizeOptions.where,
```
Comment (romain-gilliotte) [inline src/services/leaderboard-stat-getter.js:80]: doesn't that crashes when the child model is using a scope on another BelongsTo than the parent model?
It looks like the issue I had on Has-many

File: src/services/leaderboard-stat-getter.js (hunk 2 above)
Comment (larcin) [reply src/services/leaderboard-stat-getter.js:82]: Just did it, it's joined properly and the condition appears twice if provided twice in 2 separate scopes (which was intended)

**Review by romain-gilliotte — APPROVED** [review]

**Pull request merged**